src/sidebar/util/observe-element-size.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+/**
+ * Watch for changes in the size (`clientWidth` and `clientHeight`) of
+ * an element.
+ *
+ * Returns a cleanup function which should be called to remove observers when
+ * updates are no longer needed.
+ *
+ * @param {Element} element - HTML element to watch
+ * @param {(width: number, height: number) => any} onSizeChanged -
+ *   Callback to invoke with the `clientWidth` and `clientHeight` of the
```

Comment:
It occurs to me to wonder why, in the current application of this for excerpt, we need `clientWidth` as well as `clientHeight`? Symmetry? Extensibility? Or actual need I'm not thinking of?

Reply:
Actually, nevermind. The name of the module explains it 😁

Reply:
It is true that we only care about height for the initial use case, but I made it a little more generic as that was trivial to do.